### PR TITLE
Bump `wasmtime` to 6.0.0

### PR DESCRIPTION
**Pull request by koute**
This PR updates `wasmtime` to 6.0.0.

This new version includes my fix to make LTO work again, so we can finally get rid of our own `wasmtime` fork.

Also of particular note here is that `wasmtime` 6.0.0 enabled a new optimization framework which heavily regresses the contracts pallet's performance (cc @athei); I still haven't finished investigating *why* exactly this happens, but for the time being we can just revert this back to 5.0.0's behavior by changing the relevant option when initializing `wasmtime`. Otherwise the performance looks to be mostly unchanged.

### CODE REVIEW

File: client/executor/wasmtime/src/runtime.rs [hunk 1]
```diff
@@ -368,7 +373,7 @@ fn common_config(semantics: &Semantics) -> std::result::Result<wasmtime::Config,
 
 		let mut pooling_config = wasmtime::PoolingAllocationConfig::default();
 		pooling_config
-			.strategy(wasmtime::PoolingAllocationStrategy::ReuseAffinity)
+			.max_unused_warm_slots(4)
```
Comment (michalkucharczyk) [inline client/executor/wasmtime/src/runtime.rs:376]: out of curiosity: why 4?

**Review by michalkucharczyk — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2423593

File: client/executor/wasmtime/src/runtime.rs (hunk 1 above)
Comment (koute) [reply client/executor/wasmtime/src/runtime.rs:376]: It's a low enough number that's bigger than `1`, less than the default of `100` and seems to be enough for us. :P The docs for this [are here](https://docs.rs/wasmtime/latest/wasmtime/struct.PoolingAllocationConfig.html#method.max_unused_warm_slots) if you're interested. They're a little confusing, but if I'm reading them right then the value doesn't really matter as we instantiate a separate instance of `Engine` for each module, so there shouldn't be any other warm slots for the newly instantiated instance to hijack, so essentially this is just a "just-in-case" option.

**Review by athei — APPROVED** [review]
Maybe the regression with egraphs can be removed by `wasm-opt`. But lets merge this and https://github.com/paritytech/substrate/pull/13408 first. Then we can check if the regression persists in a follow up PR.

**Review by bkchr — APPROVED** [review]

File: client/executor/wasmtime/src/runtime.rs (hunk 1 above)
Comment (melekes) [reply client/executor/wasmtime/src/runtime.rs:376]: would be good to add this as a comment

**Review by melekes — APPROVED** [review]
👍

**Comment by koute** [comment]
bot merge

**Pull request merged**